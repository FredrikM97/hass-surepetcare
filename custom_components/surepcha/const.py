""" "Constants for the Sure Petcare integration."""

DOMAIN = "surepcha"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
ENTRY_ID = "entry_id"
SCAN_INTERVAL = 300
POLLING_SPEED = "polling_speed"
LOCATION_INSIDE = "location_inside"
LOCATION_OUTSIDE = "location_outside"
OPTION_DEVICES = "devices"
DEVICE_OPTION = "device_option"
OPTIONS_FINISHED = "finished"
PRODUCT_ID = "product_id"
DEVICES = "devices"
NAME = "name"
