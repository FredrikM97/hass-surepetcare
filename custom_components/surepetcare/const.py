"""TODO."""

DOMAIN = "surepetcare"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
