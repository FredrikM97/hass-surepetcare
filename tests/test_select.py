import pytest
from unittest.mock import patch
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import initialize_entry

from pytest_homeassistant_custom_component.common import (
    MockConfigEntry,
    snapshot_platform,
)


@patch("custom_components.surepetcare.PLATFORMS", [Platform.SELECT])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


@patch("custom_components.surepetcare.PLATFORMS", [Platform.SELECT])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery_missing_entities(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry_missing_entities: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry_missing_entities, mock_devices, mock_pets
    )
    await snapshot_platform(
        hass, entity_registry, snapshot, mock_config_entry_missing_entities.entry_id
    )


@patch("custom_components.surepetcare.PLATFORMS", [Platform.SELECT])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_select_set_option_and_snapshot(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await hass.async_block_till_done()

    # Iterate over all select entities
    for entity_id in hass.states.async_entity_ids("select"):
        state = hass.states.get(entity_id)
        if not state:
            continue

        # Try to get the available options from the state attributes
        options = state.attributes.get("options")
        if not options:
            continue

        # Try each option for this select
        for option in options:
            if option is None:
                continue  # Skip None options
            await hass.services.async_call(
                "select",
                "select_option",
                {"entity_id": entity_id, "option": option},
                blocking=True,
            )
            await hass.async_block_till_done()

            # Snapshot the updated state after setting each option
            updated_state = hass.states.get(entity_id)
            assert updated_state is not None
            snapshot.assert_match(updated_state)
