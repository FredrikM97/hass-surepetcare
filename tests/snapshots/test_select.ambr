# serializer version: 1
# name: test_platform_setup_and_discovery[select.katzenklappe_locking_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'UNLOCKED',
        'ALLOW_IN',
        'ALLOW_OUT',
        'LOCKED',
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.katzenklappe_locking_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Locking Mode',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'locking',
    'unique_id': '1299453-locking',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[select.katzenklappe_locking_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Katzenklappe Locking Mode',
      'options': list([
        'UNLOCKED',
        'ALLOW_IN',
        'ALLOW_OUT',
        'LOCKED',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.katzenklappe_locking_mode',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'LOCKED',
  })
# ---
# name: test_platform_setup_and_discovery[select.laundry_door_curfew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        True,
        False,
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.laundry_door_curfew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Curfew active',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'curfew_enabled',
    'unique_id': '727608-curfew_enabled',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[select.laundry_door_curfew_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Curfew active',
      'options': list([
        True,
        False,
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.laundry_door_curfew_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'False',
  })
# ---
# name: test_platform_setup_and_discovery[select.maui_matlada_old_lid_close_delay-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'FASTER',
        'NORMAL',
        'SLOWER',
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.maui_matlada_old_lid_close_delay',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Lid close delay',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'lid',
    'unique_id': '269654-lid',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[select.maui_matlada_old_lid_close_delay-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Lid close delay',
      'options': list([
        'FASTER',
        'NORMAL',
        'SLOWER',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.maui_matlada_old_lid_close_delay',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'FASTER',
  })
# ---
# name: test_platform_setup_and_discovery[select.maui_matlada_old_training_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'DISABLED',
        'STEP_1',
        'STEP_2',
        'STEP_3',
        'STEP_4',
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.maui_matlada_old_training_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Training Mode',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'training_mode',
    'unique_id': '269654-training_mode',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[select.maui_matlada_old_training_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Training Mode',
      'options': list([
        'DISABLED',
        'STEP_1',
        'STEP_2',
        'STEP_3',
        'STEP_4',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.maui_matlada_old_training_mode',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'DISABLED',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.katzenklappe_locking_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'UNLOCKED',
        'ALLOW_IN',
        'ALLOW_OUT',
        'LOCKED',
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.katzenklappe_locking_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Locking Mode',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'locking',
    'unique_id': '1299453-locking',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.katzenklappe_locking_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Katzenklappe Locking Mode',
      'options': list([
        'UNLOCKED',
        'ALLOW_IN',
        'ALLOW_OUT',
        'LOCKED',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.katzenklappe_locking_mode',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'LOCKED',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.laundry_door_curfew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        True,
        False,
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.laundry_door_curfew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Curfew active',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'curfew_enabled',
    'unique_id': '727608-curfew_enabled',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.laundry_door_curfew_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Curfew active',
      'options': list([
        True,
        False,
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.laundry_door_curfew_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'False',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.maui_matlada_old_lid_close_delay-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'FASTER',
        'NORMAL',
        'SLOWER',
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.maui_matlada_old_lid_close_delay',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Lid close delay',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'lid',
    'unique_id': '269654-lid',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.maui_matlada_old_lid_close_delay-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Lid close delay',
      'options': list([
        'FASTER',
        'NORMAL',
        'SLOWER',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.maui_matlada_old_lid_close_delay',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'FASTER',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.maui_matlada_old_training_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'DISABLED',
        'STEP_1',
        'STEP_2',
        'STEP_3',
        'STEP_4',
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.maui_matlada_old_training_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Training Mode',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'training_mode',
    'unique_id': '269654-training_mode',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[select.maui_matlada_old_training_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui matlåda (old)  Training Mode',
      'options': list([
        'DISABLED',
        'STEP_1',
        'STEP_2',
        'STEP_3',
        'STEP_4',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.maui_matlada_old_training_mode',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'DISABLED',
  })
# ---
